src/test/webviews/util.ts

```diff
@@ -0,0 +1,165 @@
+/*!
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as assert from 'assert'
+import { EventEmitter } from 'vscode'
+import { WebviewApi } from 'vscode-webview'
+import { WebviewClient, WebviewClientAgent } from '../../webviews/client'
+import { VueWebview, VueWebviewPanel, VueWebviewView } from '../../webviews/main'
+import { mock } from '../utilities/mockito'
+
+/**
+ * Mock API construct intended for low-level testing of framework code.
+ *
+ * Test code that injects their own {@link WebviewClient} should not use this.
+ * Only use this class when manually responding to messages as sent by the client agent.
+ */
+export class MockWebviewApi<T, M = any> implements WebviewApi<T> {
+    private static _instance: MockWebviewApi<any, any>
+    private postedMessages: M[] = []
+    private states: (T | undefined)[] = []
+
+    public static get instance(): MockWebviewApi<any, any> {
+        if (!this._instance) {
+            throw new Error('The mock webview API should be created by registration before using it.')
+        }
+        return this._instance
+    }
+
+    private constructor(initialState?: T) {
+        this.states.push(initialState)
+        MockWebviewApi._instance = this
+    }
+
+    public get messages(): M[] {
+        return this.postedMessages
+    }
+
+    /**
+     * Adds a mocked version of the API to `globalThis` under `vscode`.
+     */
+    public static register<U>(initialState?: U): MockWebviewApi<U, any> {
+        return ((globalThis as any).vscode = new MockWebviewApi(initialState))
+    }
+
+    public postMessage(message: any): void {
+        this.postedMessages.push(message)
+    }
+
+    public getState(): T | undefined {
+        return this.states[this.states.length - 1]
+    }
+
+    public setState<U extends T | undefined>(newState: U): U {
+        return this.states[this.states.push(newState) - 1] as U
+    }
+
+    public assertState<U extends T | undefined>(state: U, when: number = this.states.length): void {
+        assert.deepStrictEqual(this.states[when - 1], state)
+    }
+
+    public assertMessage(message: M, when: number = this.postedMessages.length): void {
+        assert.deepStrictEqual(this.postedMessages[when - 1], message)
+    }
+
+    public clear(): void {
+        this.states = []
+        this.postedMessages = []
+    }
+}
+
+/**
+ * Sets up global state for running tests
+ */
+export function registerGlobals(): void {
```

Comment:
This function dynamically loads in the required modules.
I think this can be done with mocha directly but have yet to try.